File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/Hexaghost_Model.ts
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class Hexaghost_Model extends MonsterInfo{
    public static Id = "Hexaghost"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 9)
            return 264;
        else
            return 250;
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/boss/guardian/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/boss/guardian/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/boss/guardian/skeleton.png"));
    }

    get Name(): string {
        return "怪物-六火亡魂";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/Hexaghost_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class Hexaghost_Model : MonsterInfo
    {
        public static string Id = "Hexaghost";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 9 ? 264 : 250;

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/boss/hexaghost/skeleton.atlas",
                    "Monsters/theBottom/boss/hexaghost/skeleton.json", "Monsters/theBottom/boss/hexaghost/skeleton.png");
            }
        }

        public override string Name => "怪物-六火亡魂";

        public Hexaghost_Model(Vector2 pos) : base(pos)
        {
        }
    }
}